test/tests/mendeleyImport.js

```diff
@@ -1,12 +1,34 @@
-/* global setHTTPResponse:false, sinon: false, Zotero_Import_Mendeley: false */
+/* global setHTTPResponse:false, sinon: false, Zotero_Import_Mendeley: false, HttpServer: false */
 
 describe('Zotero_Import_Mendeley', function () {
-	var server, importer;
+	var server, importer, httpd, httpdURL;
 
 	before(async () => {
 		Components.utils.import('chrome://zotero/content/import/mendeley/mendeleyImport.js');
 		importer = new Zotero_Import_Mendeley();
 		importer.mendeleyCode = 'CODE';
+
+		// real http server is used to deliver an empty pdf so that annotations can be processed during import
+		Components.utils.import("resource://zotero-unit/httpd.js");
+		const emptyPDF = await Zotero.File.getContentsFromURLAsync('resource://zotero-unit-tests/data/empty.pdf');
+		const port = 16213;
+		httpd = new HttpServer();
+		httpdURL = `http://127.0.0.1:${port}`;
+		httpd.start(port);
+		httpd.registerPathHandler('/file1.pdf',
+			{
+				handle: function (request, response) {
+					response.setStatusLine(null, 200, "OK");
+					response.setHeader('Content-Type', 'application/json', false);
+					response.write(emptyPDF);
+				}
+			});
+	});
+
+	after(function* () {
+		var defer = new Zotero.Promise.defer();
+		httpd.stop(() => defer.resolve());
+		yield defer.promise;
```

Comment:
I assume this is copied and pasted, but might as well write it how we would these days:

```javascript
after(async function () {
    await new Zotero.Promise((resolve) => {
        httpd.stop(resolve);
    });
}

Reply:
I've used a bit shorter version (below) but happy to change if that's not in a right coding style

````javascript
after(async () => new Zotero.Promise(resolve => httpd.stop(resolve)));
````

Reply:
Even better, except we usually leave these as multi-line so they're easier to extend. So let's split the difference:

```javascript
after(async () => {
    await new Zotero.Promise(resolve => httpd.stop(resolve));
}
```